src/librustdoc/html/markdown.rs

```diff
@@ -1034,12 +1034,99 @@ impl MarkdownSummaryLine<'_> {
     }
 }
 
+/// Renders a subset of Markdown in the first paragraph of the provided Markdown.
+///
+/// - *Italics*, **bold**, and `inline code` styles **are** rendered.
+/// - Headings and links are stripped (though the text *is* rendered).
+/// - HTML, code blocks, and everything else are ignored.
+///
+/// Returns a tuple of the rendered HTML string and whether the output was shortened
+/// due to the provided `length_limit`.
+fn markdown_summary_with_limit(md: &str, length_limit: Option<u16>) -> (String, bool) {
```

Comment:
Why is `length_limit` an `Option` and not just `u16`? It's only used in one place after all...

Reply:
I wanted to give the ability to render a summary without a length limit in the future.

Reply:
We have to limit it otherwise the search index size will be too huge.

Reply:
Done.